### PR TITLE
Fix apt proxy configuration on ubunt/debian

### PR DESCRIPTION
**Pull request by erdii**
**What this PR does / why we need it**: this pr fixes apt proxy configuration
before - the templated script would generate:

```
cat <<EOF | sudo tee /etc/apt/apt.conf.d/proxy.confAcquire::https::Proxy = "{{ .HTTPS_PROXY }}";
```

which broke the setup when one enabled proxy support in k1

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: 

**Does this PR introduce a user-facing change?**:

```release-note
Fix broken apt config file generation when http/https proxies are enabled. 
```

### CODE REVIEW

**Comment by erdii** [comment]
should we/I maybe think about adding an E2E test with proxies?

File: pkg/scripts/os.go [hunk 1]
```diff
@@ -55,13 +55,12 @@ EOF
...
-{{ end }}
-
+{{- if .HTTPS_PROXY }}
+Acquire::https::Proxy "{{ .HTTPS_PROXY }}";
+{{ end -}}
 {{- if .HTTP_PROXY -}}
-Acquire::http::Proxy = "{{ .HTTP_PROXY }}";
-{{ end }}
+Acquire::http::Proxy "{{ .HTTP_PROXY }}";
+{{ end -}}
```
Comment (kron4eg) [inline pkg/scripts/os.go]: this last `{{ end -}}` broke the rendering result (from bash PoV).
The rendering result is `cat <<EOF | sudo tee /etc/apt/apt.conf.d/proxy.confEOF` which caused kubeadm to be never installed and fail everything else.

Comment (erdii) [reply pkg/scripts/os.go]: thanks for the feedback - i will fix it and push again after lunch - can i retrigger the tests myself?

**Comment by erdii** [comment]
/retest

i changed the template to not strip whitespace at the end (to fix the bug I introduced) and at the beginning

File: pkg/scripts/os.go [hunk 2]
```diff
@@ -55,12 +55,11 @@ EOF
...
-Acquire::https::Proxy = "{{ .HTTPS_PROXY }}";
-{{ end }}
-
+{{ if .HTTPS_PROXY -}}
+Acquire::https::Proxy "{{ .HTTPS_PROXY }}";
+{{ end -}}
 {{- if .HTTP_PROXY -}}
-Acquire::http::Proxy = "{{ .HTTP_PROXY }}";
+Acquire::http::Proxy "{{ .HTTP_PROXY }}";
 {{ end }}
```
Comment (kron4eg) [inline pkg/scripts/os.go]:
```
{{- if .HTTPS_PROXY }}
Acquire::https::Proxy "{{ .HTTPS_PROXY }}";
{{- end }}
{{- if .HTTP_PROXY }}
Acquire::http::Proxy "{{ .HTTP_PROXY }}";
{{- end }}
```

here you can play with it: https://play.golang.org/p/v_uScHqYcKH

**Comment by erdii** [comment]
@kron4eg tests are passing again - how do I continue from here? also:

> should we/I maybe think about adding an E2E test with proxies?

File: pkg/scripts/os.go (hunk 2 above)
Comment (erdii) [reply pkg/scripts/os.go]: thx - your solution looks and behaves cleaner - i will take that

**Comment by kron4eg** [comment]
>adding an E2E test with proxies

That will complicate out current e2e matrix, which is already out of control. It's a task for unknown future.

**Comment by erdii** [comment]
ok sounds fair to me! thx for clarification

**Comment by kron4eg** [comment]
@erdii please take a look at https://github.com/kubermatic/kubeone/blob/master/CONTRIBUTING.md#release-notes-block section (this PR currently doesn't feature a release-note block, that's why it has automatic `do-not-merge/release-note-label-needed` label)

**Pull request merged**